parcels/fieldset.py

```diff
@@ -444,25 +464,13 @@ def from_netcdf(cls, filenames, variables, dimensions, indices=None, fieldtype=N
                                 processedGrid *= filenames[procvar][dim] == filenames[var][dim]
                     if processedGrid:
                         grid = fields[procvar].grid
-                        # ==== check here that the dims of field_chunksize are the same ==== #
-                        if grid.master_chunksize is not None:
-                            res = False
-                            if (isinstance(chunksize, tuple) and isinstance(grid.master_chunksize, tuple)) or (isinstance(chunksize, dict) and isinstance(grid.master_chunksize, dict)):
-                                res |= functools.reduce(lambda i, j: i and j, map(lambda m, k: m == k, chunksize, grid.master_chunksize), True)
-                            else:
-                                res |= (chunksize == grid.master_chunksize)
-                            if res:
-                                grid_chunksize = grid.master_chunksize
-                                logger.warning_once("Trying to initialize a shared grid with different chunking sizes - action prohibited. Replacing requested field_chunksize with grid's master chunksize.")
-                            else:
-                                raise ValueError("Conflict between grids of the same fieldset chunksize and requested field chunksize as well as the chunked name dimensions - Please apply the same chunksize to all fields in a shared grid!")
```

Comment:
Now, I am not entirely happy with this solution out of performance reasons. We did agree that, if the chunksize is 'auto' or the chunksizes are different, the fields get different grids. Now, as we compare full sub-dictionaries in line 450, it leads us to a situation where, even if U an V essentially chunk their fields the same sizes, if the NetCDF dimensions and their chunking only differs by the name (see MITgcm -> YC/YG and XC/XG), then they will get different grids. Even if, numerically, they could get the same grid. It's not satisfying from an efficieny viewpoint. 

What this restriction does do is lead to trimmed, less code, indeed.

Reply:
Good point, @CKehl. I have now pushed a suggestion (plus unit test) that checks if the values (not the dimension names) are the same for the different chunking dictionary entries. Would that be a solution?

Reply:
i'll look at it - but probably not today. Today I need to wrap up something else.